tests/integration/ha_tests/test_self_healing.py

```diff
@@ -250,18 +253,22 @@ async def test_full_cluster_restart(
     # they come back online they operate as expected. This check verifies that we meet the criteria
     # of all replicas being down at the same time.
     assert await all_db_processes_down(ops_test, process), "Not all units down at the same time."
+    if process == PATRONI_PROCESS:
+        awaits = []
+        for unit in ops_test.model.applications[app].units:
+            awaits.append(update_restart_condition(ops_test, unit, ORIGINAL_RESTART_CONDITION))
```

Comment:
Snap does not set a custom restart delay and it is more reliable to not depend on timeouts.